linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -115,6 +115,22 @@ void Process ()
 				ProcessQueue ();
 				ProcessVirtualMethods ();
 			}
+
+			// deal with [TypeForwardedTo] pseudo-attributes
+			foreach (AssemblyDefinition assembly in _context.GetAssemblies ()) {
+				if (!assembly.MainModule.HasExportedTypes)
+					continue;
+
+				foreach (var exported in assembly.MainModule.ExportedTypes) {
+					if (!exported.IsForwarder)
+						continue;
+					var type = exported.Resolve ();
+					if (!Annotations.IsMarked (type))
+						continue;
+					Annotations.Mark (exported);
+					Annotations.Mark (assembly.MainModule);
```

Comment:
If a type forwarder needs to be kept, then the module/assembly containing the type forwarder also needs to be kept. In .NET Core we have some assemblies that only have type forwarders and we ran into a case where the assembly was incorrectly deleted.